crates/bevy_ecs/src/system/query.rs

```diff
@@ -1431,6 +1431,96 @@ impl<'w, 's, D: QueryData, F: QueryFilter> Query<'w, 's, D, F> {
+    ///
+    /// ```rust
+    /// # use bevy_ecs::prelude::*;
+    /// # use bevy_ecs::system::QueryLens;
+    /// #
+    /// # #[derive(Component)]
+    /// # struct A(usize);
+    /// #
+    /// # #[derive(Component)]
+    /// # struct B(usize);
+    /// #
+    /// # let mut world = World::new();
+    /// #
+    /// # world.spawn((A(10), B(5)));
+    /// #
+    /// fn reusable_function(lens: &mut QueryLens<&A>) {
+    ///     assert_eq!(lens.query().single().0, 10);
+    /// }
+    ///
+    /// // We can use the function in a system that takes the exact query.
+    /// fn system_1(mut query: Query<&A>) {
+    ///     reusable_function(&mut query.into_query_lens());
+    /// }
+    ///
+    /// // We can also use it with a query that does not match exactly
+    /// // by transmuting it.
+    /// fn system_2(mut query: Query<(&mut A, &B)>) {
+    ///     let mut lens = query.transmute_lens::<&A>();
+    ///     reusable_function(&mut lens);
+    /// }
+    ///
+    /// # let mut schedule = Schedule::default();
+    /// # schedule.add_systems((system_1, system_2));
+    /// # schedule.run(&mut world);
+    /// ```
+    ///
+    /// ## Allowed Transmutes
+    ///
+    /// Besides removing parameters from the query, you can also
+    /// make limited changes to the types of paramters.
+    ///
+    /// * Can always add/remove `Entity`
+    /// * `Ref<T>` <-> `&T`
+    /// * `&mut T` -> `&T`
+    /// * `&mut T` -> `Ref<T>`
+    /// * [`EntityMut`](crate::world::EntityMut) -> [`EntityRef`](crate::world::EntityRef)
+    ///    
+    pub fn transmute_lens<NewD: QueryData>(&mut self) -> QueryLens<'_, NewD> {
+        self.transmute_lens_filtered::<NewD, ()>()
+    }
+
+    /// Equivalent to [`Self::transmute_lens`] but also includes a [`QueryFilter`] type.
+    ///
+    /// Note that the lens will iterate the same tables and archetypes as the original query. This means that
+    /// additional archetypal query terms like [`With`](crate::query::With) and [`Without`](crate::query::Without)
+    /// will not necessarily be respected and non-archetypal terms like [`Added`](crate::query::Added) and
+    /// [`Changed`](crate::query::Changed) will only be respected if they are in the type signature.
+    pub fn transmute_lens_filtered<NewD: QueryData, NewF: QueryFilter>(
+        &mut self,
+    ) -> QueryLens<'_, NewD, NewF> {
+        // SAFETY: There are no other active borrows of data from world
+        let world = unsafe { self.world.world() };
+        let state = self.state.transmute_filtered::<NewD, NewF>(world);
+        QueryLens {
+            world: self.world,
+            state,
+            last_run: self.last_run,
+            this_run: self.this_run,
+            force_read_only_component_access: self.force_read_only_component_access,
+        }
+    }
+
+    /// Gets a [`QueryLens`] with the same accesses as the existing query
+    pub fn into_query_lens(&mut self) -> QueryLens<'_, D> {
+        self.transmute_lens()
+    }
```

Comment:
```suggestion
    /// Gets a [`QueryLens`] with the same accesses as the existing query
    pub fn as_lens(&mut self) -> QueryLens<'_, D> {
        self.transmute_lens()
    }
```

This doesn't consume self so I don't think this should be prefixed with `into`